### PR TITLE
[h2v][doc] Repush optimization

### PR DESCRIPTION
**Pull request by FelixGV**
## Summary
Memory allocation reduction for repush

## How was this PR tested?
Internal CI and manual test on an OOMing job.

## Does this PR introduce any user-facing changes?
No

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java [hunk 1]
```diff
@@ -51,10 +51,9 @@ protected Optional<VeniceWriterMessage> extract(
     if (!valueIterator.hasNext()) {
       throw new VeniceException("There is no value corresponding to key bytes: " + ByteUtils.toHexString(keyBytes));
     }
-    List<KafkaInputMapperValue> mapperValues = getMapperValues(valueIterator);
     return isChunkingEnabled()
-        ? extractChunkedMessage(keyBytes, mapperValues)
-        : extractNonChunkedMessage(keyBytes, mapperValues);
+        ? extractChunkedMessage(keyBytes, getMapperValues(valueIterator))
```
Comment (huangminchn) [inline clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java:55]: Would you like to leave a TODO message about how to optimize memory usage for chunked stores?

**Review by huangminchn — DISMISSED** [review]
Thanks for the quick turnaround Felix!

File: clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java (hunk 1 above)
Comment (FelixGV) [reply clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java:55]: Hoping to take a stab at it tomorrow, but wanted to keep the current change surgical.

Comment (FelixGV) [reply clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java:55]: Marked `getMapperValues` function as deprecated and left a TODO.

**Review by huangminchn — APPROVED** [review]

**Pull request merged**